javascript/javascript3/week2/warmup/script.js

```diff
@@ -0,0 +1,116 @@
+document.body.appendChild(latestBadMv);
+fetch(moviesUrl)
+    .then(Response => Response.json())
+    .then(movies => {
+        const badMovies = movies.filter(movie => movie.rating < 4);
+        badMovies.forEach(mv => {
+            const myLi = document.createElement("li");
+            myLi.style.fontSize = "10px";
+            myLi.innerHTML = `${mv.title} with ${mv.rating} rating`;
+            myOl.appendChild(myLi);
+        });
+        //Creat an array of bad movies since year 2000
+        const latestBadMovied = badMovies.filter(movie => movie.year >= 2000);
+        latestBadMovied.forEach(mv => {
+            const latestBadMvLi = document.createElement("li");
+            latestBadMvLi.style.fontSize = "10px";
+            latestBadMvLi.innerHTML = `${mv.title} in year ${mv.year}`;
+            latestBadMv.appendChild(latestBadMvLi);
+        });
+    })
+
+//Promise that resolves after set time & use it with async/await
+async function myDelayMessage(resolveAfter) {
+    await new Promise((resolve, reject) => {
+        if (typeof resolveAfter !== "number") {
+            reject()
+        }
+        setTimeout(function() {
+            resolve()
+        }, resolveAfter * 1000)
+    })
+}
+myDelayMessage(8)
+    .then(() => {
+        console.log("I am called asynchronously"); // logged out after 8 seconds
+    })
+    .catch(() => {
+        console.log("Second must be a number!")
+    });
+
+//Rewrite time
+//Rewrite setTimeout
+function setTimeoutPromise(time) {
+    return new Promise(resolve => {
+        setTimeout(function() {
+            resolve();
+        }, time)
+    })
+}
+setTimeoutPromise(3000).then(() => {
+    console.log("Called after 3 seconds ")
+})
+
+//Rewrite navigator.geolocation.getCurrentPosition
+
+let currentLocation = document.createElement("p")
+document.body.appendChild(currentLocation);
+
+function getCurrentPosition() {
+    return new Promise((resolve, reject) => {
+        if (navigator.geolocation) {
+            resolve();
+        } else {
+            reject();
+        }
+    })
+}
+getCurrentPosition()
+    .then(() => {
+        navigator.geolocation.getCurrentPosition(position => {
+            let latitude = position.coords.latitude;
+            let longitude = position.coords.longitude;
+            currentLocation.innerHTML = `Latitude; ${latitude} 
+           <br>
+           Longitude; ${longitude}`
+        })
+    })
+    .catch(() => {
+        currentLocation.innerHTML = "Geolocation is not supported by this browser."
+    })
+
+//Fetching and waiting with Promise
+function getDogsNames() {
+    return new Promise((resolve) => {
+        setTimeout(() => {
+            resolve();
+        }, 3000)
+    })
+}
+getDogsNames()
+    .then(() => {
+        fetch("https://dog.ceo/api/breeds/list/all")
+            .then(Response => Response.json())
+            .then(data => {
+                console.log(Object.keys(data.message))
+            })
```

Comment:
again you put the logic into the `then` rather than into the function `getDogsNames`

instead of nesting the fetch inside the then, could you create a timeoutPromise(), getDogsNames() and chain them like this:

```
timeoutPromise(3000)
.then(getDogsNames)
.then(res => {
    console.log(res);
})
```

does that make sense?